sdk/src/main/java/ly/count/android/sdk/ConnectionQueue.java

```diff
@@ -429,6 +429,29 @@ public void sendCrashReport(@NonNull final String crashData) {
         tick();
     }
 
+    /**
+     * Send a direct request to server
+     *
+     * @throws IllegalStateException if context, app key, store, or server URL have not been set
+     */
+    //public void sendCrashReport(String error, boolean nonfatal, boolean isNativeCrash, final Map<String, Object> customSegmentation) {
+    public void sendDirectRequest(@NonNull final String requestData) {
+        checkInternalState();
+        L.d("[Connection Queue] sendDirectRequest");
+
+        if (!consentProvider.getConsent(Countly.CountlyFeatureNames.events)) {
+            L.d("[Connection Queue] request ignored, consent not given");
+            return;
+        }
+
+        final String data = prepareCommonRequestData()
+            + "&events=" + UtilsNetworking.urlEncodeString(requestData);
```

Comment:
Rework this closer to:

//master.count.ly/i?a=123&b=ert&device_id=a
    public void sendDirectRequest(Map<String, String> vals) {

        StringBuilder data = prepareCommonRequestData();

        foreach(KeyValuePair<String, String> value:vals) {
            data.append("&");
            data.append(UtilsNetworking.urlEncodeString(value.key()));
            data.append("=");
            data.append(UtilsNetworking.urlEncodeString(value.value()));
        }

        addRequestToQueue(data);
        tick();
    }